pootle/apps/pootle_store/models.py

```diff
@@ -1750,6 +1764,18 @@ def update(self, overwrite=False, store=None, only_newer=False):
                     self.get_max_unit_revision())
                 )
 
+    def increment_unsynced_unit_revision(self, update_revision):
+        filter_by = {'store': self}
+        filter_by.update({'revision__gt': self.last_sync_revision,
+                          'revision__lt': update_revision})
+        count = 0
+        for unit in Unit.objects.filter(**filter_by).exclude(state=OBSOLETE):
```

Comment:
If instead of `Unit.objects` we use `self.unit_set` we can avoid passing the store on `filter_by`.

Perhaps it should be also considered moving the exclude condition to `filter_by` as `'state__gt': OBSOLETE,` since that excludes obsolete units from the queryset.